src/app/info/InfoPage.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import WikiItems from "@/app/info/WikiItems";
+
+import { Supabase } from "@/supabaseUtils";
+import { DbWikiRow } from "@/databaseUtils";
+import supabase from "@/supabaseClient";
+import Error from "next/error";
+
+interface QueryType {
+  data: DbWikiRow | null;
+  error: Error | null;
+}
+
+const getTopWikiRow = async (supabase: Supabase): Promise<QueryType> => {
+    const query = (await supabase.from("wiki").select("*").limit(1).single()) as QueryType;
+    return query;
+};
+
+const InfoPage: React.FC = () => {
+    const [topRow, setTopRow] = useState<DbWikiRow | null>(null); 
+    try {
+      useEffect(() => {
+        const fetchData: () => Promise<void> = async () => {
+            const temp: QueryType = await getTopWikiRow(supabase);
+            if (temp.error) {
+              throw temp.error;
+            } else if(temp.data) {
+              setTopRow(temp.data);
+            }
+        };
+        fetchData();
+      }, []);
+    } catch (e: any) {
+      return (
+      <>
+        <h3>Error Occured</h3>
+        <p>{e}</p>
+      </>
+      );
+    }
```

Comment:
Although this is valid code, wrapping hooks is in general a bit of an anti-pattern.
The pattern elsewhere in the codebase is to have two state variables, one for the error and one for the data, set them both in useEffect, then render conditionally based on them. `src/app/clients/clientsTable/ClientsPage.tsx` has some good examples for comparison

Reply:
switched to SSR so no hooks anymore